.github/workflows/tests.yml

```diff
@@ -0,0 +1,51 @@
+name: Tests
+
+on:
+  push:
+  pull_request:
+  schedule:
+    - cron: '0 0 * * *'
+
+jobs:
+  tests:
+    runs-on: ubuntu-latest
+
+    strategy:
+      fail-fast: true
+      matrix:
+        php: [7.2, 7.3, 7.4, 8.0]
+        stability: [prefer-lowest, prefer-stable]
+
+    name: PHP ${{ matrix.php }} - ${{ matrix.stability }}
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - name: Setup PHP
+        uses: shivammathur/setup-php@v2
+        with:
+          php-version: ${{ matrix.php }}
+          extensions: dom, curl, libxml, mbstring, zip
+          coverage: none
+
+      - name: Install dependencies
+        run: composer update --${{ matrix.stability }} --prefer-dist --no-interaction --no-progress
+
+#      - name: Run PHP CS Fixer
+#        if: ${{ matrix.php != 8.0 }}
+#        run: |
+#          composer require friendsofphp/php-cs-fixer:^2.15 --dev --prefer-dist --no-interaction --no-progress --update-with-all-dependencies
+#          vendor/bin/php-cs-fixer fix -v --dry-run --using-cache=no
+
+      - name: Run PHPStan
+        run: vendor/bin/phpstan analyze --level max -c phpstan.neon src
+
+      - name: Execute tests
+        run: vendor/bin/phpunit --verbose --coverage-text --coverage-clover=coverage.clover
+
+      - name: Code coverage
+        if: ${{ github.ref == 'refs/heads/master' && matrix.php != 8.0 }}
```

Comment:
I'm not 100% sure if the `php` check is still needed here. Might also be something you want to try out yourself.